src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -177,4 +186,136 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
+                    if (!student.getGroups().contains(course.getStudentGroupName())) {
+                        userService.addUserToGroup(student, course.getStudentGroupName());
+                    }
+                    continue;
+                }
+                // 2) if we cannot find the student, we use the registration number and try to find the student in the (TUM) LDAP, create it in the Artemis DB and in a potential
+                // external user management system
+                optionalStudent = userService.createUserFromLdap(registrationNumber);
+                if (optionalStudent.isPresent()) {
+                    var student = optionalStudent.get();
+                    exam.addUser(student);
+                    // the newly created student needs to get the rights to access the course, otherwise the student cannot access the exam (within the course)
+                    userService.addUserToGroup(student, course.getStudentGroupName());
+                    continue;
+                }
+                // 3) if we cannot find the user in the (TUM) LDAP, we report this to the client
+                log.warn("User with registration number " + registrationNumber + " not found in Artemis user database and not found in (TUM) LDAP");
+            }
+            catch (Exception ex) {
+                log.warn("Error while processing user with registration number " + registrationNumber + ": " + ex.getMessage(), ex);
+            }
+
+            notFoundStudentsDtos.add(studentDto);
+        }
+        examRepository.save(exam);
+        return ResponseEntity.ok().body(notFoundStudentsDtos);
+    }
+
+    /**
+     * DELETE /courses/:courseId/exams/:examId/students/:studentLogin : Remove one single given user (based on the login) from the students of the exam so that the student cannot access the exam any more
+     *
+     * @param courseId     the id of the course
+     * @param examId       the id of the exam
+     * @param studentLogin the login of the user who should lose student access
+     * @return empty ResponseEntity with status 200 (OK) or with status 404 (Not Found)
+     */
+    @DeleteMapping(value = "/courses/{courseId}/exams/{examId}/students/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> removeStudentFromExam(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable String studentLogin) {
+        log.debug("REST request to remove {} as student from exam : {}", studentLogin, examId);
+        var course = courseService.findOne(courseId);
+        var instructorOrAdmin = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, instructorOrAdmin)) {
+            return forbidden();
+        }
+        var exam = examService.findOneWithRegisteredUsers(examId);
+        if (!course.equals(exam.getCourse())) {
+            return conflict();
+        }
```

Comment:
We could use `checkCourseAndExamAccess()` from `ExamAccessService` here.

Reply:
👍 I will change it in a follow-up PR